src/web/app/pages-instructor/instructor-search-page/instructor-search-bar/instructor-search-bar.component.ts

```diff
@@ -9,8 +18,16 @@ import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
   styleUrls: ['./instructor-search-bar.component.scss'],
 })
 export class InstructorSearchBarComponent implements OnInit {
-  @Input() searchKey: string = '';
-  @Output() searched: EventEmitter<string> = new EventEmitter<string>();
+
+  @Input() searchParams: SearchParams = {
+    searchKey: '',
+    isSearchForStudents: true,
+    isSearchForComments: false
+  }
+
+  @Output() searched: EventEmitter<SearchParams> = new EventEmitter();
+
+  @Output() searchParamsChange: EventEmitter<SearchParams> = new EventEmitter();
```

Comment:
Why the event is not used?

Reply:
Which event is not used?

`searched` is used when user clicks/enter the search button

`searchParamsChange` is used in the two way binding for `searchParams` in `instructor-search-page.component`

Reply:
The 2-way binding does not happen magically if you do not fire the event `searchParamsChange.emit(newSearchParams)`